### PR TITLE
Build poetry

### LINKED ISSUE #45: import ._version
`from ._version import __version__` is currently used to get the version number for the interim datasets. However, it does not work if the package is installed from github directly (`pip install git+https://github.com/atmdrops/pydropsonde.git#egg=pydropsonde`). I am playing around with poetry at the moment to see if I can somehow fix that, but if you have any ideas, please let me know.

### PR DESCRIPTION
**Pull request by hgloeckner**
This is the preparation to make pydropsonde a real pip-package

- I head to remove typhon as a dependency. I really tried, but I couldn't figure out how it works without breaking. Since we only need some physics functions, I guess it is also fine to use them (it solves our negative iwv problem on the fly)
- fixes #45

### CODE REVIEW

**Comment by hgloeckner** [comment]
it also fixes the problems with the tests that we haven't put in an issue yet @ninarobbins ^^'

**Comment by ninarobbins** [comment]
Sounds good to not use syphon, it's also nice to see exactly how things are calculated

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -1,51 +1,43 @@
...
+        uses: abatilo/actions-poetry@v2
+      - name: Setup a local virtual environment (if no poetry.toml file)
+        run: |
+          poetry config virtualenvs.create true --local
+          poetry config virtualenvs.in-project true --local
+      - uses: actions/cache@v3
+        name: Define a cache for the virtual environment based on the dependencies lock file
+        with:
+          path: ./.venv
+          key: venv-${{ hashFiles('poetry.lock') }}
```
[the remark this answers is not in the document]
Comment (tmieslinger) [reply .github/workflows/python-package.yml:29]: For the release a commited lock file would be good to add, too.

**Review by tmieslinger — COMMENTED** [review]
Adding poetry is a really nice thing!
I tested building the dependencies and there is one comment on the poetry.lock file form our discussion earlier.

File: pytest.ini
Comment (d70-t) [inline pytest.ini:1]: You may want to consider configuring [pytest in pyproject.toml](https://docs.pytest.org/en/7.1.x/reference/customize.html#pyproject-toml)

Comment (hgloeckner) [reply pytest.ini:1]: I just removed the file... I think it is not needed

File: .github/workflows/python-package.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (hgloeckner) [reply .github/workflows/python-package.yml:26]: it's not so easy to tell poetry to ignore the lock file... so I just added another section

**Pull request merged**